guidelines/static-websites.md

```diff
@@ -0,0 +1,137 @@
+  GitHub repository inside infra-managed organizations is the best way for us
+  to ensure that. Even though we'd prefer for all the repositories to be public
+  it's not a requirement.
+* **The website must be built and deployed with Travis CI.**  
+  We have custom tooling built around hosting static websites on our infra,
+  and at the moment they only work with Travis CI. If you need different CI
+  services we can try to adapt our tools.
+* **The website must reach an A+ grade on the
+  [Mozilla Observatory](https://observatory.mozilla.org/).**  
+  Browsers have multiple security features toggleable only through HTTP
+  response headers, and those features enhance users' privacy and prevent
+  exploits from working. An A+ grade on the Observatory indicates all the
+  important headers are correctly set.
+* **The website can be hosted on either GitHub Pages or Amazon S3 (in the
+  rust-lang account), and must sit behind CloudFront.**  
+  Our custom tooling needs CloudFront to be in front of your website in order
+  to apply extra HTTP headers, and GitHub Pages or Amazon S3 are platforms we
+  vetted for security and reliability.
+
+## Adding custom headers
+
+One of the requirements for having a static website hosted by the
+infrastructure team is to reach an A+ grade on the [Mozilla
+Observatory](https://observatory.mozilla.org/), and that requires custom
+headers to be set. Unfortunately neither GitHub Pages nor Amazon S3 allows
+setting up them, but we have pieces of infrastructure setup to work around the
+limitation.
+
+To setup custom headers a file named `_rustinfra_config.json` needs to be
+included in the generated website. This example content includes all the
+headers needed to reach grade B on the Observatory (to reach grade A+ a Content
+Security Policy needs to be added):
+
+```json
+{
+    "headers": {
+        "Strict-Transport-Security": "max-age=63072000",
+        "X-Content-Type-Options": "nosniff",
+        "X-Frame-Options": "DENY",
+        "X-XSS-Protection": "1; mode=block",
+        "Referrer-Policy": "no-referrer, strict-origin-when-cross-origin"
+    }
+}
+```
+
+If you’re using Jekyll as the build tool for the website you also need to
+explicitly whitelist the file in the `_config.yml`, otherwise Jekyll will
+ignore it since it starts with a `_`:
+
+```yaml
+include:
+  - _rustinfra_config.json
+```
+
+## Deployment guide
+
+### Configuring AWS
+
+Create a CloudFront web distribution and set the following properties:
+
+- **Origin Domain Name:** rust-lang.github.io/repo-name
+- **Origin Protocol Policy:** HTTPS Only
+- **Viewer Protocol Policy:** Redirect HTTP to HTTPS
+- **Lambda Function Association:**
+    - **Viewer Response:** arn:aws:lambda:us-east-1:890664054962:function:static-websites:1
+- **Alternate Domain Names:** your-subdomain-name.rust-lang.org
+- **SSL Certificate:** Custom SSL Certificate
+    - You will need to request the certificate for that subdomain name through ACM
```

Comment:
May be worth noting here that the preferred method of ACM requests is DNS configuration so we don't get a bunch of emails that someone has to remember to click.

Reply:
Added.